pandas/tests/frame/test_to_csv.py

```diff
@@ -1108,3 +1108,9 @@ def test_to_csv_with_dst_transitions(self):
             df.to_pickle(path)
             result = pd.read_pickle(path)
             assert_frame_equal(result, df)
+
+    def test_to_csv_empty_frame(self):
+        # GH12048
+        actual = read_csv(StringIO('A,B'), dtype=str)
+        expected = pd.DataFrame({'A': [], 'B': []}, dtype=str).reset_index(drop=True)
```

Comment:
construct the expected to exactly match, e.g. with the appropriate index type, names etc